sdk/tables/azure-data-tables/azure/data/tables/_serialize.py

```diff
@@ -34,7 +34,10 @@ def _get_match_headers(etag, match_condition):
 
 def _prepare_key(keyvalue):
     """Duplicate the single quote char to escape."""
-    return keyvalue.replace("'", "''")
+    try:
+        return keyvalue.replace("'", "''")
+    except AttributeError:
+        raise TypeError('PartitionKey or RowKey must be of type string.')
```

Comment:
Could we please add a unit test for this?

Reply:
That's a good point! Let me add in another PR as this one has auto-merged.